frontend/src/app/components/auth/auth.service.ts

```diff
@@ -155,7 +155,7 @@ export class AuthService {
   }
 
   isAuthenticated(): boolean {
-    return this._cookie.get('token') !== null;
+    return this._cookie.check('token') && this._cookie.get('token') !== null;
```

Comment:
```suggestion
    return this._cookie.check('token') && this._cookie.get('token')
```
Normalement, le `!==null` est implicite.